app/viewtils/EvidenceFileUploadsHelper.scala

```diff
@@ -16,19 +16,48 @@
 
 package viewtils
 
-import models.{Mode, UserRequest}
 import models.upload.UploadJourney
+import models.{Mode, UserRequest}
 import play.api.i18n.Messages
 import play.twirl.api.Html
+import uk.gov.hmrc.govukfrontend.views.Aliases.{InsetText, Text}
+import uk.gov.hmrc.govukfrontend.views.html.components.GovukInsetText
 import views.html.components.upload.uploadList
 
 import javax.inject.Inject
 
-class EvidenceFileUploadsHelper @Inject()(uploadList: uploadList){
+class EvidenceFileUploadsHelper @Inject()(govukInsetText: GovukInsetText,
+                                          uploadList: uploadList) {
 
   def displayContentForFileUploads(uploadedFiles: Seq[(UploadJourney, Int)], mode: Mode)(implicit messages: Messages, request: UserRequest[_]): Seq[Html] = {
     uploadedFiles.map(uploadWithIndex =>
       uploadList(uploadWithIndex._1.reference, uploadWithIndex._1.uploadDetails.get.fileName, uploadWithIndex._2 + 1, mode)
     )
   }
+
+  def duplicateDocumentInset(uploadedFiles: Seq[(UploadJourney, Int)])(
+    implicit messages: Messages, request: UserRequest[_]): Option[Html] = {
+
+    val mappedDuplicates = uploadedFiles.groupBy(_._1.uploadDetails.map(_.checksum))
+
+    if (!mappedDuplicates.exists(_._2.size > 1)) {
```

Comment:
I'm just thinking about when we have WAITING uploads? If we look at the JS version, we could have multiple uploads that are WAITING and we might get None -> Seq(..., ...) which could cause a logical error - maybe if we filter the uploads?

Reply:
The uploads are already being filtered here 
```
          val uploadedFileNames: Seq[UploadJourney] = previousUploadsState.fold[Seq[UploadJourney]](Seq.empty)(_.filter( upload =>
            upload.fileStatus == UploadStatusEnum.READY || upload.fileStatus == UploadStatusEnum.DUPLICATE))
```
in the controller before being passed into the to the helper function. So I don't think any marked as waiting would get through to this part of the function. Unless I'm misunderstanding things 😅

Reply:
Doh! Yes you are right! It was Friday 😉